src/NuGet.Core/NuGet.ProjectModel/DependencyGraphSpec.cs

```diff
@@ -130,6 +130,18 @@ public void AddProject(PackageSpec projectSpec)
             _projects.Add(projectUniqueName, projectSpec);
         }
 
+        /// <summary>
+        /// Add all restorable projects to the restore list.
```

Comment:
From the callers point of view AddRestore and AddRestoreForAllProjects have similar intent, marking projects in the dg file for restore. Ensure would technically be more correct here, but I don't feel that the caller needs to care about that implementation detail, with Add it is easier to discover.